voxpupuli-puppet-lint-plugins.gemspec

```diff
@@ -15,26 +15,26 @@ Gem::Specification.new do |s|
 
   # pull in puppet-lint 2.5.0; required for Ruby 3 support
   s.add_runtime_dependency 'puppet-lint', '>= 2.5.0'
-  s.add_runtime_dependency 'puppet-lint-absolute_classname-check', '>= 2.0.0'
-  s.add_runtime_dependency 'puppet-lint-anchor-check'
-  s.add_runtime_dependency 'puppet-lint-classes_and_types_beginning_with_digits-check'
-  s.add_runtime_dependency 'puppet-lint-file_ensure-check'
-  s.add_runtime_dependency 'puppet-lint-leading_zero-check'
-  s.add_runtime_dependency 'puppet-lint-legacy_facts-check'
-  s.add_runtime_dependency 'puppet-lint-lookup_in_parameter-check'
-  s.add_runtime_dependency 'puppet-lint-manifest_whitespace-check'
-  s.add_runtime_dependency 'puppet-lint-optional_default-check'
-  s.add_runtime_dependency 'puppet-lint-param-docs'
-  s.add_runtime_dependency 'puppet-lint-params_empty_string-check'
-  s.add_runtime_dependency 'puppet-lint-param-types'
-  s.add_runtime_dependency 'puppet-lint-resource_reference_syntax'
-  s.add_runtime_dependency 'puppet-lint-strict_indent-check'
-  s.add_runtime_dependency 'puppet-lint-top_scope_facts-check'
-  s.add_runtime_dependency 'puppet-lint-topscope-variable-check'
-  s.add_runtime_dependency 'puppet-lint-trailing_comma-check'
-  s.add_runtime_dependency 'puppet-lint-unquoted_string-check'
-  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase'
-  s.add_runtime_dependency 'puppet-lint-version_comparison-check'
+  s.add_runtime_dependency 'puppet-lint-absolute_classname-check', '>= 3.0.1', '< 4.0.0'
+  s.add_runtime_dependency 'puppet-lint-anchor-check', '>= 1.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-classes_and_types_beginning_with_digits-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-file_ensure-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-leading_zero-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-legacy_facts-check', '>= 1.0.4', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-lookup_in_parameter-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-manifest_whitespace-check', '>= 0.2.6', '< 1.0.0'
+  s.add_runtime_dependency 'puppet-lint-optional_default-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-param-docs', '>= 1.7.6', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-params_empty_string-check', '~> 1'
+  s.add_runtime_dependency 'puppet-lint-param-types', '>= 0.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-resource_reference_syntax', '>= 1.1.0', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-strict_indent-check', '>= 2.0.8', '< 3.0.0'
+  s.add_runtime_dependency 'puppet-lint-top_scope_facts-check', '>= 1.0.1', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-topscope-variable-check', '>= 1.1.0', '< 2.0.0'
+  s.add_runtime_dependency 'puppet-lint-trailing_comma-check', '>= 0.4.3', '< 1.0.0'
+  s.add_runtime_dependency 'puppet-lint-unquoted_string-check', '>= 2.1.0', '< 3.0.0'
+  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase', '>= 1.2.0', '< 2.0.0'
```

Comment:
```suggestion
  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase', '~> 1.2.0'
```

Reply:
but that isn't the same I think? doesn't `~> 1.2.0` only allow `1.2.X`?

Reply:
https://bundler.io/guides/gemfile.html
> The specifier ~> has a special meaning, best shown by example. ~> 2.0.3 is identical to >= 2.0.3 and < 2.1. ~> 2.1 is identical to >= 2.1 and < 3.0. ~> 2.2.beta will match prerelease versions like 2.2.beta.12. ~> 0 is identical to >= 0.0 and < 1.0.

Reply:
Oh yes, I meant
```suggestion
  s.add_runtime_dependency 'puppet-lint-variable_contains_upcase', '~> 1.2'
```